stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp

```diff
@@ -264,9 +265,15 @@ Result FsMitmService::OpenBisStorage(Out<std::shared_ptr<IStorageInterface>> out
             if (bis_partition_id == BisStorageId_Boot0) {
                 storage = std::make_shared<IStorageInterface>(new Boot0Storage(bis_storage, this->title_id));
             } else if (bis_partition_id == BisStorageId_Prodinfo) {
+                bool has_blank_cal0_flag = true; // See PR details
+
                 /* PRODINFO should *never* be writable. */
                 if (is_sysmodule || has_cal0_read_flag) {
                     storage = std::make_shared<IStorageInterface>(new ROProxyStorage(bis_storage));
+                } else if (has_blank_cal0_flag) {
+                    FsFile file = Utils::GetBlankProdinfoFileHandle();
+                    
+                    storage = std::make_shared<IStorageInterface>(new FileStorage(std::make_shared<ProxyFile>(&file)));
```

Comment:
This probably needs special handling to not leak domain IDs.

Reply:
How would I go about that?